File: Exercism Codes/Java/DnDCharacter.java
import java.util.Random;

class DnDCharacter {
    private Random random = new Random();
    private int strength = ability();
    private int dexterity = ability();
    private int constitution = ability();
    private int intelligence = ability();
    private int wisdom = ability();
    private int charisma = ability();
    
    int ability() {
        return random.ints(4, 1, 7).sorted().skip(1).sum();
    }
    int modifier(int input) {
        return Math.floorDiv(input - 10, 2);
    }
    int getStrength() {
        return strength;
    }
    int getDexterity() {
        return dexterity;
    }
    int getConstitution() {
        return constitution;
    }
    int getIntelligence() {
        return intelligence;
    }
    int getWisdom() {
        return wisdom;
    }
    int getCharisma() {
        return charisma;
    }
    int getHitpoints() {
        return 10 + modifier(constitution);
    }
}

File: Exercism Codes/JavaScript/dnd-character.js
export const abilityModifier = (n) => {
    if(n > 18) throw new Error('Ability scores can be at most 18');
    if(n < 3) throw new Error('Ability scores must be at least 3');
    return Math.floor((n - 10) / 2);
};
export class Character {
    constructor() {
        this.str = Character.rollAbility();
        this.dex = Character.rollAbility();
        this.con = Character.rollAbility();
        this.int = Character.rollAbility();
        this.wis = Character.rollAbility();
        this.cha = Character.rollAbility();
    }

    static rollAbility() {
        return [...new Array(4).keys()]
      .map(n => Math.ceil(Math.random() * 6))
      .toSorted()
      .slice(1)
      .reduce((a, c) => a + c, 0);
    }
    get strength() {
        return this.str;
    }
    get dexterity() {
        return this.dex;
    }
    get constitution() {
        return this.con;
    }
    get intelligence() {
        return this.int;
    }
    get wisdom() {
        return this.wis;
    }
    get charisma() {
        return this.cha;
    }
    get hitpoints() {
        return 10 + abilityModifier(this.constitution);
    }
}
